docs/data/data-grid/pagination/CursorPaginationGrid.js

```diff
@@ -37,38 +48,94 @@ export default function CursorPaginationGrid() {
     }
   };
 
+  const paginationMetaRef = React.useRef();
+
+  // Memoize to avoid flickering when the `hasNextPage` is `undefined` during refetch
+  const paginationMeta = React.useMemo(() => {
+    if (
+      hasNextPage !== undefined &&
+      paginationMetaRef.current?.hasNextPage !== hasNextPage
+    ) {
+      paginationMetaRef.current = { hasNextPage };
+    }
+    return paginationMetaRef.current;
+  }, [hasNextPage]);
+
   React.useEffect(() => {
-    if (!isLoading && pageInfo?.nextCursor) {
+    if (!isLoading && nextCursor) {
       // We add nextCursor when available
-      mapPageToNextCursor.current[paginationModel.page] = pageInfo?.nextCursor;
+      mapPageToNextCursor.current[paginationModel.page] = nextCursor;
     }
-  }, [paginationModel.page, isLoading, pageInfo?.nextCursor]);
+  }, [paginationModel.page, isLoading, nextCursor]);
 
   // Some API clients return undefined while loading
   // Following lines are here to prevent `rowCountState` from being undefined during the loading
-  const [rowCountState, setRowCountState] = React.useState(
-    pageInfo?.totalRowCount || 0,
-  );
+  const [rowCountState, setRowCountState] = React.useState(totalRowCount || 0);
   React.useEffect(() => {
```

Comment:
Use `React.useMemo` instead, similar to https://github.com/mui/mui-x/pull/12474#discussion_r1553170263?

Reply:
I'm afraid switching from state to memoized variable here might not give the best results.
1. When controlling the `rowCount`: It will block the default behavior of the Grid i.e. setting the `rowCount` to a finite ( > 0 ) number when `hasNextPage` becomes `false` (as we can't pass `onRowCountChange`)
2. When initializing the `rowCount`: It will make the switching between `known`, `unknown`, and `estimated` row counts not work as expected as the grid will not react to the updation of `initialState.pagination.rowCount`
Wdyt?

Reply:
Makes sense to me 👍🏻